src/GazeboThruster.cpp

```diff
@@ -192,6 +192,7 @@ void GazeboThruster::updateBegin(common::UpdateInfo const& info)
             thruster != thrusters.end(); ++thruster)
     {
         physics::LinkPtr link = model->GetLink( thruster->name );
+        link->AddLinkForce( thruster->effort*gazebo::math::Vector3::UnitX);
         link->AddLinkForce( thruster->effort*thruster->direction, thruster->position );
```

Comment:
Wouldn't it be less convoluted to transfer the thruster force to the CoG and then apply the added mass ?

What I still don't understand is that you apply the exact same effort in both places. Unless I'm missing something ...